# A Robust Marker Gene Selection by Harmonizing Expression Levels and Positivity Rates in Single-Cell Resolution Transcriptome
[![python~=3.8](https://img.shields.io/badge/python-3.8-brightgreen)](https://www.python.org/)
[![License: GPL3.0](https://img.shields.io/badge/License-GPL3.0-yellow)](https://opensource.org/license/gpl-3-0/)

Cell type classification is the most import step for the analysis of single-cell and spatial transcriptome data, conep (**con**sistency between gene **e**xpression and cell **p**ositive rate) provides an accurate and stable marker gene identification. Conep is fast and better at finding some biologically meaningful marker genes.

![image](https://github.com/zhuqianhua/Conep/tree/main/assets/workflow_of_conep.jpg)

[Quick Example](https://github.com/zhuqianhua/Conep/tree/main/tutorials/conep_tutorials.ipynb) - [Citation](https://github.com/zhuqianhua/Conep)

# Installation

```
pip install git+https://github.com/zhuqianhua/Conep.git
```

# Usage

The [conep tutorials](https://github.com/zhuqianhua/Conep/tree/main/tutorials/conep_tutorials.ipynb) provides a quick-start guide based pbmc3k dataset.

# Enviroments
- python>=3.8.0
- numpy>=1.18.0
- pandas>=0.25.1
- scanpy>=1.9.0
- scipy>=1.9.0
- anndata>=0.8.0
- scikit-learn>=0.19.0

# Question

For questions about the code and tutorial, please contact Qianhua ZHU, zhuqianhua@genomics.cn.

# Citation
If corep is useful for your research, please consider citing.